### PR TITLE
add mirror health checking

### PR DESCRIPTION
**Pull request by sctb512**
Backport nydus mirror health checking patch to stable/v0.3

Related to https://github.com/containerd/nydus-snapshotter/pull/217

Signed-off-by: Bin Tang <tangbin.bin@bytedance.com>

### CODE REVIEW

File: config/daemonconfig.go [hunk 1]
```diff
@@ -61,8 +61,12 @@ type DaemonConfig struct {
 }
 
 type MirrorConfig struct {
-	Host    string            `json:"host,omitempty"`
-	Headers map[string]string `json:"headers,omitempty"`
+	Host                string            `json:"host,omitempty"`
+	Headers             map[string]string `json:"headers,omitempty"`
+	AuthThrough         bool              `json:"auth_through,omitempty"`
+	HealthCheckInterval int               `json:"health_check_interval,omitempty"`
+	FailureLimit        uint8             `json:"failure_limit,omitempty"`
```
Comment (imeoer) [inline config/daemonconfig.go:68]: Why HealthCheckInterval uses  `uint8` but HealthCheckInterval uses `int`?

Comment (sctb512) [reply config/daemonconfig.go:68]: Sorry, the `HealthCheckInterval` is refer to `CheckInterval` for `proxy`. Should we use `int` for both of them?

Comment (sctb512) [reply config/daemonconfig.go:68]: I also find other items which have different types between `nydus` and `nydus-snapshotter`. Should we fix it together?

nydus:
```rust
pub timeout: u32,
pub connect_timeout: u32,
pub retry_limit: u8,
```

nydus-snapshotter:
```golang
Timeout        int `json:"timeout,omitempty"`
ConnectTimeout int `json:"connect_timeout,omitempty"`
RetryLimit     int `json:"retry_limit,omitempty"`
```

Comment (sctb512) [reply config/daemonconfig.go:68]: And the `FailureLimit ` references to the `RetryLimit` in `proxy`.

Comment (imeoer) [reply config/daemonconfig.go:68]: @sctb512 Thanks for explaining, it's ok, we can make snapshotter consistent with nydusd.

File: config/daemonconfig.go [hunk 2]
```diff
@@ -61,8 +61,12 @@ type DaemonConfig struct {
 }
 
 type MirrorConfig struct {
-	Host    string            `json:"host,omitempty"`
-	Headers map[string]string `json:"headers,omitempty"`
+	Host                string            `json:"host,omitempty"`
+	Headers             map[string]string `json:"headers,omitempty"`
+	AuthThrough         bool              `json:"auth_through,omitempty"`
+	HealthCheckInterval uint64            `json:"health_check_interval,omitempty"`
```
Comment (changweige) [inline config/daemonconfig.go:67]: Um...  uint64 is a considerable number. Is it necessary?

Comment (changweige) [reply config/daemonconfig.go:67]: However, it's not a blocker, let's merge it now

**Review by changweige — APPROVED** [review]
LGTM

File: config/daemonconfig.go (hunk 2 above)
Comment (sctb512) [reply config/daemonconfig.go:67]: No, It is not necessary, just to adapt the type in `image-service`.

**Pull request merged**